torchao/quantization/quantize_/workflows/float8/float8_tensor.py

```diff
@@ -177,18 +177,29 @@ def to_float8(
         block_size = get_block_size(hp_tensor.shape, granularity)
         block_size = list(block_size)
 
+        kernel_choice = None
         # for per row quantization and kernel_preference default setting, we'll use triton kernel for best performance
         if (
             kernel_preference == KernelPreference.AUTO
             and _is_fbgemm_genai_gpu_available()
-            and (
-                tuple(block_size)
-                == (1,) * (hp_tensor.ndim - 1) + (hp_tensor.shape[-1],)
-            )
+            and is_sm_at_least_90()
+            and isinstance(granularity, PerRow)
+            and float8_dtype == torch.float8_e4m3fn
+            and hp_value_lb is None
         ):
-            assert float8_dtype == torch.float8_e4m3fn, (
-                f"Only torch.float8_e4m3fn is supported, got: {float8_dtype}"
+            # optimized path for auto and per row quantization
+            kernel_choice = "triton"
```

Comment:
let's make it clear what is "triton" and how it is related to `KernelPreference`

Reply:
IMO we shouldn't have anything in here which is not in `KernelPreference`, it's too confusing.  Not confusing options would be either extending `KernelPreference`, or creating a new thing if you need more granularity.

Reply:
we can add a triton kernel preference as well I think